examples/ping_pong.c

```diff
@@ -0,0 +1,343 @@
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program; if
+ * not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ *
+ * BXI Low Level Team
+ *
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/mman.h>
+#include <unistd.h>
+#include <stdatomic.h>
+
+#include "portals4.h"
+#include "portals4_bxiext.h"
+
+/*
+ * Example how to use logical interface to communicate within two interfaces
+ */
+
+int main(void)
+{
+	int ret;
+	char msg[PTL_EV_STR_SIZE];
+	ptl_process_t id;
+	ptl_handle_ni_t nih1;
+	ptl_handle_ni_t nih2;
+	ptl_handle_eq_t eqh1;
+	ptl_index_t pti1;
+	ptl_handle_md_t mdh1;
+	ptl_handle_le_t leh1;
+	ptl_handle_eq_t eqh2;
+	ptl_index_t pti2;
+	ptl_handle_md_t mdh2;
+	ptl_handle_le_t leh2;
+	ptl_event_t ev;
+	ptl_process_t rank0;
+	ptl_process_t rank1;
+	ptl_process_t *mapping;
+	ptl_le_t le1;
+	ptl_le_t le2;
+	ptl_md_t md1;
+	ptl_md_t md2;
+	char init_data[50];
+	char *data1 = "Bonjour interface 2 !";
+	char *data2 = "Bonjour interface 1 !";
+
+	mapping = (ptl_process_t *)mmap(NULL, getpagesize(), PROT_READ | PROT_WRITE,
+					MAP_ANONYMOUS | MAP_SHARED, 0, 0);
+	atomic_int *is_ready =
+		&((atomic_int *)(mapping))[2 * (sizeof(ptl_process_t) / sizeof(int)) + 1];
+	*is_ready = ATOMIC_VAR_INIT(0);
+
+	atomic_int *is_ready_bis =
+		&((atomic_int *)(mapping))[2 * (sizeof(ptl_process_t) / sizeof(int)) +
+					   sizeof(is_ready) + 1];
+	*is_ready_bis = ATOMIC_VAR_INIT(0);
+
+	/*fork in two process for our two interfaces*/
+	int res = fork();
+
+	rank0.rank = 0;
+	rank1.rank = 1;
+
+	ret = PtlInit();
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlInit failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		return 1;
+	}
+
+	/* interface 1 */
+	if (res != 0) {
+		ret = PtlNIInit(PTL_IFACE_DEFAULT, PTL_NI_NO_MATCHING | PTL_NI_LOGICAL, PTL_PID_ANY,
+				NULL, NULL, &nih1);
+		if (ret != PTL_OK) {
+			fprintf(stderr, "PtlNIInit failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+			return 1;
+		}
+
+		ret = PtlGetPhysId(nih1, &id);
+		if (ret != PTL_OK) {
+			fprintf(stderr, "PtlGetId failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+			return 1;
+		} else {
+			mapping[0] = id;
+			int val = atomic_load(is_ready) + 1;
+			atomic_store(is_ready, val);
+		}
+
+		while (atomic_load(is_ready) < 2) {
+		}
+
+		/*Set mapping from logical to physical id*/
+
+		ret = PtlSetMap(nih1, 2, mapping);
+		if (ret != PTL_OK) {
+			fprintf(stderr, "PtlSetMap failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+			return 1;
+		} else {
+			int val = atomic_load(is_ready_bis) + 1;
+			atomic_store(is_ready_bis, val);

```

Comment:
I think we can simply do `*is_ready_bis += 1`, it should be atomic as `is_ready_bis` is declared as an atomic_int

The same should be done on the other atomics.
The issue with this code is that we have a race condition that could lead to both processes being stuck forever